File: Single-Number/SN.py
class Solution(object):

    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        numDict = {}

        for i in nums:
            try:
                numDict[i] += 1
            except:
                numDict[i] = 1

        for key, value in numDict.items():
            if value == 1:
                return key

File: Single-Number/SN.rs
pub fn single_number(nums: Vec<i32>) -> i32 {
    use std::collections::HashMap;

    let mut tbl: HashMap<i32, usize> = HashMap::new();
    for num in nums {
        *tbl.entry(num).or_insert(0) += 1;
    }
    *tbl.iter().filter(|x| *x.1 == 1).next().unwrap().0
}

fn main() {
    dbg!(single_number(vec![1, 1, 2]));
}
